controllers/trainingjob/trainingjob_controller.go

```diff
@@ -322,6 +364,12 @@ func (r *Reconciler) updateStatusWithAdditional(ctx reconcileRequestContext, tra
 	jobStatus.SecondaryStatus = trainingJobSecondaryStatus
 	jobStatus.Additional = additional
 
+	// Check if we should update debugRuleEvaluationStatus
+	if ctx.TrainingJobDescription != nil && len(ctx.TrainingJobDescription.DebugRuleEvaluationStatuses) > 0 {
```

Comment:
Why are we duplicating this functionality? We should have only [one piece of code responsible](https://en.wikipedia.org/wiki/Single_responsibility_principle) for updating the debug rule statuses.

Reply:
I think `updateStatusWithAdditional` update the status of object. In this function if you notice we populate all the trainingjob attribute and update it, hence the debugrule status

Reply:
Fixed